src/parquet/column_reader.h

```diff
@@ -130,7 +141,61 @@ class PARQUET_EXPORT ColumnReader {
   ::arrow::MemoryPool* pool_;
 };
 
-// API to read values from a single column. This is the main client facing API.
+namespace internal {
+
+static inline void DefinitionLevelsToBitmap(const int16_t* def_levels,
+                                            const int64_t num_def_levels,
+                                            const int16_t max_definition_level,
+                                            const int16_t max_repetition_level,
+                                            int64_t* values_read, int64_t* null_count,
+                                            uint8_t* valid_bits,
+                                            const int64_t valid_bits_offset) {
```

Comment:
Do you know if C/C++ compilers will generate better code if parameters are marked const? If they can be analyzed as invariant then perhaps there is no difference

Reply:
Actually, after reading around I think it's probably *better* to declare these `const`. You can't modify the value (local to the function of course) which makes reasoning about it easier. So, this can be left as is.